scripts/babbage/script-context-equivalance-test.sh

```diff
@@ -0,0 +1,198 @@
+# by a plutus script, it must have a datahash. We also need collateral tx inputs so we split the utxo
+# in order to accommodate this.
+
+
+plutusscriptaddr=$($CARDANO_CLI address build --payment-script-file "$plutusscriptinuse"  --testnet-magic "$TESTNET_MAGIC")
+readonlyaddress=addr_test1vz3t3f2kgy2re66tnhgxc4t8jgylw2cqfnxdwlrq9agfmtstxxkm5
+
+mkdir -p "$WORK"
+
+utxoaddr=$($CARDANO_CLI address build --testnet-magic "$TESTNET_MAGIC" --payment-verification-key-file "$UTXO_VKEY")
+
+$CARDANO_CLI query utxo --address "$utxoaddr" --cardano-mode --testnet-magic "$TESTNET_MAGIC" --out-file $WORK/utxo-1.json
+cat $WORK/utxo-1.json
+
+txin=$(jq -r 'keys[]' $WORK/utxo-1.json)
+lovelaceattxin=$(jq -r ".[\"$txin\"].value.lovelace" $WORK/utxo-1.json)
+lovelaceattxindiv6=$(expr $lovelaceattxin / 6)
+
+$CARDANO_CLI query protocol-parameters --testnet-magic "$TESTNET_MAGIC" --out-file $WORK/pparams.json
+
+$CARDANO_CLI transaction build \
+  --babbage-era \
+  --cardano-mode \
+  --testnet-magic "$TESTNET_MAGIC" \
+  --change-address "$utxoaddr" \
+  --tx-in "$txin" \
+  --tx-out "$readonlyaddress+$lovelaceattxindiv6" \
+  --tx-out "$plutusscriptaddr+$lovelaceattxindiv6" \
+  --tx-out-datum-hash "$scriptdatumhash" \
+  --tx-out "$utxoaddr+$lovelaceattxindiv6" \
+  --protocol-params-file "$WORK/pparams.json" \
+  --out-file "$WORK/create-datum-output.body"
+
+$CARDANO_CLI transaction sign \
+  --tx-body-file $WORK/create-datum-output.body \
+  --testnet-magic "$TESTNET_MAGIC" \
+  --signing-key-file $UTXO_SKEY \
+  --out-file $WORK/create-datum-output.tx
+
+# SUBMIT
+$CARDANO_CLI transaction submit --tx-file $WORK/create-datum-output.tx --testnet-magic "$TESTNET_MAGIC"
+echo "Pausing for 5 seconds..."
+sleep 5
+
+# Step 2
+# After "locking" the tx output at the script address, we can now can attempt to spend
+# the "locked" tx output below.
+
+$CARDANO_CLI query utxo --address $plutusscriptaddr --testnet-magic "$TESTNET_MAGIC" --out-file $WORK/plutusutxo.json
+
+plutusutxotxin=$(jq -r 'keys[]' $WORK/plutusutxo.json)
+
+$CARDANO_CLI query utxo --address $utxoaddr --cardano-mode --testnet-magic "$TESTNET_MAGIC" --out-file $WORK/utxo-2.json
+cat $WORK/utxo-2.json
+txinCollateral=$(jq -r 'keys[0]' $WORK/utxo-2.json)
+
+
+dummyaddress=addr_test1vpqgspvmh6m2m5pwangvdg499srfzre2dd96qq57nlnw6yctpasy4
+
+lovelaceatplutusscriptaddr=$(jq -r ".[\"$plutusutxotxin\"].value.lovelace" $WORK/plutusutxo.json)
+
+#Get read only reference input
+$CARDANO_CLI query utxo --address "$readonlyaddress" --cardano-mode \
+  --testnet-magic "$TESTNET_MAGIC" --out-file $WORK/read-only-ref-input-utxo.json
+readonlyrefinput=$(jq -r 'keys[0]' $WORK/read-only-ref-input-utxo.json)
+
+
+
+# We need to generate a dummy redeemer in order to create a txbody from which we can generate
+# a tx and then derive the correct redeemer. TODO: Why? The cli demands you specify a redeemer (double check this)
```

Comment:
Does this todo need addressing? I think we know the answer. (same for minting bash script)

Reply:
Nop, I'll remove the TODO